flocker/control/schema/types.yml

```diff
@@ -59,7 +59,17 @@ definitions:
         external:
           type: integer
     uniqueItems: true
-    
+
+  environment:
+    title: "Container environment variables"
+    description: "Mapping of label and value pairs for environment variables"
+    type: object
+    patternProperties:
+      "^.{1,256}$":
```

Comment:
It's for both and I now have JSON schema tests to show that. I'm not aware of what characters might not be allowed in an environment variable so I don't know what other constraints I could put on this; I used the same pattern for generic properties (JSON schema equivalent of an arbitrary dictionary) as we use in our schema for dataset metadata.